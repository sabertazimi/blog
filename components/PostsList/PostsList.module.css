a.link.link.link.link.link {
  @apply ml-2 inline-block align-bottom text-3xl font-extrabold;
  @apply transform-gpu text-primary transition hover:translate-x-4 hover:text-secondary focus:translate-x-4 focus:text-secondary;
}
