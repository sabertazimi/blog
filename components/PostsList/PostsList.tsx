import Link from '@components/Link'
import { List } from '@components/Lists'
import type { PostMeta } from '@types'
import styles from './PostsList.module.css'

interface Props {
  posts: PostMeta[]
}

function PostsList({ posts }: Props): JSX.Element {
  return (
    <List
      className="mt-6"
      dataSource={posts}
      renderItem={({ slug, title }) => (
        <List.Item>
          <List.Item.Meta
            title={(
              <Link href={`/post/${slug}`} className={styles.link}>
                {title}
              </Link>
            )}
          />
        </List.Item>
      )}
    />
  )
}

export default PostsList
