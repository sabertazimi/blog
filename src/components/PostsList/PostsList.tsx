import type { PostMetaType } from '@types';
import { List } from 'antd';
import { Link } from 'gatsby';
import React from 'react';

interface Props {
  posts: PostMetaType[];
}

const PostsList = ({ posts }: Props): JSX.Element => (
  <List
    className="mt-6"
    dataSource={posts}
    renderItem={({ slug, title }) => (
      <List.Item>
        <List.Item.Meta
          title={
            <Link
              className="text-blue-400 transition text-span-lg transform-gpu hover:translate-x-4"
              to={`${slug}`}
            >
              {title}
            </Link>
          }
        />
      </List.Item>
    )}
  />
);

export default PostsList;
