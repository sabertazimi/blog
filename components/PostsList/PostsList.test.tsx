import mockData from '@mocks/data'
import { render } from '@utils'
import { axe } from 'jest-axe'
import PostsList from './PostsList'

describe('postsList', () => {
  const mockPosts = mockData.posts

  it('should render correctly (snapshot)', () => {
    const { container } = render(<PostsList posts={mockPosts} />)

    expect(container).toMatchSnapshot()
  })

  it('should render accessibility guidelines (AXE)', async () => {
    const { container } = render(<PostsList posts={mockPosts} />)

    const a11y = await axe(container)

    expect(a11y).toHaveNoViolations()
  })
})
