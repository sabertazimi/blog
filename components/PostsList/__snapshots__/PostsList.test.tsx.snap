// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postsList should render correctly (snapshot) 1`] = `
<div>
  <div
    class="ant-list ant-list-split mt-6"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <ul
          class="ant-list-items"
        >
          <li
            class="ant-list-item"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  <a
                    class="link"
                    href="/post/1BasicNotes"
                  >
                    1 Basic Notes
                  </a>
                </h4>
              </div>
            </div>
          </li>
          <li
            class="ant-list-item"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  <a
                    class="link"
                    href="/post/2BasicNotes"
                  >
                    2 Basic Notes
                  </a>
                </h4>
              </div>
            </div>
          </li>
          <li
            class="ant-list-item"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  <a
                    class="link"
                    href="/post/3BasicNotes"
                  >
                    3 Basic Notes
                  </a>
                </h4>
              </div>
            </div>
          </li>
          <li
            class="ant-list-item"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  <a
                    class="link"
                    href="/post/4BasicNotes"
                  >
                    4 Basic Notes
                  </a>
                </h4>
              </div>
            </div>
          </li>
          <li
            class="ant-list-item"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  <a
                    class="link"
                    href="/post/5BasicNotes"
                  >
                    5 Basic Notes
                  </a>
                </h4>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
